src/util/utils.ts

```diff
@@ -297,40 +297,23 @@ export const getYesterdayDateRoundDownHour = () => {
 type AsyncFunction = () => Promise<any>
 
 export async function asyncWaterfall(asyncFuncs: AsyncFunction[], timeoutMs: number = 5000): Promise<any> {
```

Comment:
Change of plans: Let's just ESLint ignore this function for now. It's tricky.

---

This `asyncWaterfall` refactor seems completely wrong. Since each function is (presumably) a network fetch, we don't want to invoke them unless necessary. This means a simple `Promise.race` situation won't work - we need some way to know if the first promise has succeeded, so we can avoid triggering the subsequent fetches in the first place.

There may still be a way to incorporate `Promise.race` as the core of the algorithm, if we use timeouts and flags to coordinate the different "threads" of execution, so we shut down leftover timers when we are done, and avoid triggering fetches we don't need.

I would like to see `asyncWaterfall` implemented in its own PR, along with tests, in one of the currency repos. Then we can copy-paste it over to the GUI.